addons/sale_margin/tests/test_sale_margin.py

```diff
@@ -0,0 +1,43 @@
+# -*- coding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from openerp.tests import common
+from datetime import datetime
+
+
+class TestSaleMargin(common.TransactionCase):
+
+    def setUp(self):
+        super(TestSaleMargin, self).setUp()
+        self.SaleOrder = self.env['sale.order']
+
+        self.product_uom = self.env.ref('product.product_uom_unit')
+        self.product = self.env.ref('product.product_product_24')
+        self.partner = self.env.ref('base.res_partner_4')
+        self.partner_invoice_address = self.env.ref('base.res_partner_address_7')
+        self.pricelist = self.env.ref('product.list0')
+
+    def test_00_sale_margin(self):
```

Comment:
test case should be cover proper business process
-> setup cost price in product
-> make sale order with sale price
-> check margin of that order